### PR TITLE
Feature/7640 penrollment deletion

### PR DESCRIPTION
**Pull request by ceiroa**
Fixes W-007640: What should we do with the related Affiliation when a Program Enrollment record is deleted?

I have added an additional second custom setting to allow the admin to specify which status to change the related Affiliation to.

### CODE REVIEW

File: src/objects/Hierarchy_Settings__c.object [hunk 1]
```diff
@@ -15,6 +15,28 @@
         <unique>false</unique>
     </fields>
     <fields>
+        <fullName>Affl_ProgEnroll_Del__c</fullName>
+        <defaultValue>false</defaultValue>
+        <description>Determines is an Affiliation will be automatically deleted when the related Program Enrollment is deleted. Otherwise the status will be changed to Former.</description>
```
Comment (kbromer) [inline src/objects/Hierarchy_Settings__c.object]: If "former" the default value? Or is it pulling from something else? 

Also, "Determined if" not "Determines is"

Comment (ceiroa) [inline src/objects/Hierarchy_Settings__c.object]: It's pulling it from the setting right below: Affl_ProgEnroll_Del_Status__c. We set it to "Former" during installation, but then the admin can change it.

Yep, that's a typo. I wanted to say "Determines if".

**Comment by ceiroa** [comment]
Hey @kbromer, let us know if you'd like to merge this, or if I should ask @daveperry. Thanks.

File: src/classes/TDTM_DefaultConfig.cls [hunk 2]
```diff
@@ -110,7 +110,7 @@ public without sharing class TDTM_DefaultConfig {
         // Affiliation management on Program Enrollment record creation
         handlers.add(new Trigger_Handler__c(Active__c = true, Asynchronous__c = false, 
               Class__c = 'PREN_Affiliation_TDTM', Load_Order__c = 1, Object__c = 'Program_Enrollment__c', 
-              Trigger_Action__c = 'BeforeInsert;BeforeUpdate'));
```
Comment (kbromer) [inline src/classes/TDTM_DefaultConfig.cls]: Moving from beforeUpdate to AfterDelete?  I don't see handling for either though in the changes above, did I miss it?

File: src/classes/PREN_Affiliation_TDTM.cls [hunk 3]
```diff
@@ -47,27 +47,71 @@ public class PREN_Affiliation_TDTM extends TDTM_Runnable {
...
         }
         return dmlWrapper;
     }
+    
+    private void createRelatedAffls(List<SObject> newlist, TDTM_Runnable.Action triggerAction) {
+        List<Affiliation__c> newAffls = new List<Affiliation__c>();
+        for (SObject so : newlist) {
+            Program_Enrollment__c enrollment = (Program_Enrollment__c)so;
+            
+            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
```
Comment (ceiroa) [inline src/classes/PREN_Affiliation_TDTM.cls]: @kbromer BeforeInsert check is here.

File: src/classes/PREN_Affiliation_TDTM.cls [hunk 4]
```diff
@@ -47,27 +47,71 @@ public class PREN_Affiliation_TDTM extends TDTM_Runnable {
...
+            //We don't need to add these enrollment records to dmlWrapper because we are in BeforeInsert.
+        }
+    }
+    
+    private List<ID> getDeletedPEsAfflIDs(List<SObject> oldlist, TDTM_Runnable.Action triggerAction) {
+        List<ID> afflsToUpdateDelIDs = new List<ID>();
+        for (SObject so : oldlist) {
+            Program_Enrollment__c oldEnrollment = (Program_Enrollment__c)so;
+            
+            if(triggerAction == TDTM_Runnable.Action.AfterDelete) {
```
Comment (ceiroa) [inline src/classes/PREN_Affiliation_TDTM.cls]: AfterDelete check is here.

We just weren't using BeforeUpdate.

**Pull request merged**